docs/source/api/notebooks/Notebook_4_Customized_Models.ipynb

```diff
@@ -0,0 +1,456 @@
+    "$$\\beta^r_i = \\dfrac{exp(h^{r}_i \\cdot \\phi)}{\\sum_{k \\in \\{1, \\dots, \\mathscr{R}\\}} exp(h^{k}_i \\cdot \\phi)}$$\n",
+    "\n",
+    "$$h_i = \\sum_{r \\in \\mathscr{R}}{\\beta^r_i \\times h_i^{r}}$$, where\n",
+    "\n",
+    "$$h_i^{r} = LeakyReLU(\\sum_{j\\in \\mathcal{N}^{r}_{(i)}} \\alpha^{r}_{i,j} W^{r} h_j^{r})$$\n",
+    "\n",
+    "In this notebook, we will implement this `Cross Relation Attention Enhanced RGAT (CRAE_GAT)`, fit it into GraphStorm model architecture, and run training and inference by using existing pipelines.\n",
+    "\n",
+    "### Prerequsites\n",
+    "\n",
+    "- GraphStorm. Please find [more details on installation of GraphStorm](https://graphstorm.readthedocs.io/en/latest/install/env-setup.html#setup-graphstorm-with-pip-packages).\n",
+    "- ACM data that has been created according to the **[Notebook 0: Data Prepare](https://graphstorm.readthedocs.io/en/latest/api/notebooks/Notebook_0_Data_Prepare.html)**, and is stored in the `./acm_gs_1p/` folder.\n",
+    "- Installation of supporting libraries, e.g., matplotlib."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Recap GraphStorm Model Architecture\n",
+    "\n",
+    "As explained in the **[Notebook 3: Model Variants](https://graphstorm.readthedocs.io/en/latest/api/notebooks/Notebook_3_Model_Variants.html)**, a GraphStorm model normally contains four modules:\n",
+    "\n",
+    "- An input encoder that converts input node features to the embeddings with hidden dimensions.\n",
+    "- A GNN encoder that takes the embeddings from the input layer and performs message passing computation.\n",
+    "- A decoder that is task sepcific, e.g., the `EntityClassifier` for classification tasks.\n",
+    "- A loss function that matches specific tasks, e.g., the `ClassifyLossFunc`.\n",
+    "\n",
+    "Given this architecture, it is clear that we only need to build a **customized GNN encoder** that implements the `CREA_GAT` variant and leave the other modules untouch."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. CRAE_GAT Variant Encoder Implementation\n",
+    "\n",
+    "To build a customized GNN encoder, we can refer to the behaviors of GraphStorm's GNN encoders, like `graphstorm.model.RelationalGATEncoder`, which extends the `graphstorm.model.GraphConvEncoder` and `graphstorm.model.GSgnnGNNEncoderInterface` classes, and implementing the required methods.\n",
+    "\n",
+    "Code in the below cells include a layer module, named `Crae_GatLayer`, that fulfills the CRAE_GAT functions in one layer of GNN, and an encoder module, named `Crae_GatEncoder`, that is compatible to GraphStorm APIs. \n",
```

Comment:
```suggestion
    "Code in the below cells include a layer module, named `Crae_GatLayer`, that fulfills the CRAE_GAT functions in one layer of GNN, and an encoder module, named `Crae_GatEncoder`, that follows GraphStorm APIs. \n",
```

Reply:
changed to `that extends from graphstorm.model.GSgnnNodeModel API.`